### ISSUE TITLE
[BUG] `NameError: name '__assert_armored__' is not defined` while resolving ForwardRef's

### ISSUE THREAD

**Issue by goraje** [dashingsoft/pyarmor#1287]
```
$ pyarmor gen --obf-code 2 -O armored -i src/foo
INFO     Python 3.10.11
INFO     Pyarmor 8.2.1 (group), 006005, R****t
INFO     Platform darwin.aarch64
```
This problem is one that I could only reproduce, while obfuscating a package with `--obf-code 2`.

Root directory structure:
```
.
├── MANIFEST.in
├── Makefile
├── setup.py
└── src
    └── foo
        ├── __init__.py
        ├── main.py
        └── models.py

3 directories, 6 files
```

MANIFEST.in
```
recursive-include src/foo/pyarmor_runtime_006005 *
```
setup.py:
```python
#!/usr/bin/env python
from setuptools import find_packages, setup


setup(
    name="foo",
    description="Foo package",
    author="John Doe",
    author_email="jondoe@jondoe.com",
    version="0.1.0",
    packages=find_packages(where="src"),
    package_dir={"": "src"},
    include_package_data=True,
    python_requires=">=3.10",
    install_requires=["pydantic", "fastapi", "uvicorn"],
    entry_points={
        "console_scripts": [
            "foo_run = foo.main:run"
        ]
    }
)
```

Makefile:
```
PACKAGE_NAME := "foo"
PACKAGE_DIR := "src"
PACKAGE_BKP := "src_bkp"
ARMORED_DIR := "armored"

default:
    @echo "Package:" $(PACKAGE_NAME)
    @echo "Running pyarmor"
    pyarmor gen --obf-code 2 -O $(ARMORED_DIR) -i $(PACKAGE_DIR)/$(PACKAGE_NAME)
    @echo "Syncing src"
    cp -r $(PACKAGE_DIR) $(PACKAGE_BKP)
    rsync -avh $(ARMORED_DIR)/ $(PACKAGE_DIR)
    @echo "Build wheel"
    python setup.py bdist_wheel
    @echo "Cleanup"
    rm -rf .pyarmor $(ARMORED_DIR) $(PACKAGE_DIR)
    mv $(PACKAGE_BKP) $(PACKAGE_DIR)
```

src/foo/main.py:
```python
from __future__ import annotations

import uvicorn
from fastapi import FastAPI
from fastapi.responses import JSONResponse

from foo import models

app = FastAPI()


@app.get("/foo")
def get_foo(data: models.Data) -> JSONResponse:
    return JSONResponse(content={"text": data.content.text})

def run() -> None:
    uvicorn.run(app)
```

src/foo/models.py:
```python
from __future__ import annotations

from pydantic import BaseModel

__all__ = ["Data"]

class Data(BaseModel):
    content: Content

class Content(BaseModel):
    text: str

Data.update_forward_refs()
```

src/foo/__init__.py: empty

Then I obfuscate, build and install the package with:
```
make
pip install dist/foo-0.1.0-py3-none-any.whl
```

and run the entry point with:
```
foo_run
```

which produces an error:
```
Traceback (most recent call last):
  File "/***/venv/bin/foo_run", line 5, in <module>
    from foo.main import run
  File "<frozen foo.main>", line 3, in <module>
  File "<frozen foo.main>", line 13, in <module>
  File "/***/venv/lib/python3.10/site-packages/fastapi/routing.py", line 661, in decorator
    self.add_api_route(
  File "/***/venv/lib/python3.10/site-packages/fastapi/routing.py", line 600, in add_api_route
    route = route_class(
  File "/***/venv/lib/python3.10/site-packages/fastapi/routing.py", line 446, in __init__
    self.dependant = get_dependant(path=self.path_format, call=self.endpoint)
  File "/***/venv/lib/python3.10/site-packages/fastapi/dependencies/utils.py", line 292, in get_dependant
    endpoint_signature = get_typed_signature(call)
  File "/***/venv/lib/python3.10/site-packages/fastapi/dependencies/utils.py", line 252, in get_typed_signature
    typed_params = [
  File "/***/venv/lib/python3.10/site-packages/fastapi/dependencies/utils.py", line 257, in <listcomp>
    annotation=get_typed_annotation(param.annotation, globalns),
  File "/***/venv/lib/python3.10/site-packages/fastapi/dependencies/utils.py", line 268, in get_typed_annotation
    annotation = evaluate_forwardref(annotation, globalns, globalns)
  File "pydantic/typing.py", line 66, in pydantic.typing.evaluate_forwardref
    'MutableSequence',
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/typing.py", line 694, in _evaluate
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
NameError: name '__assert_armored__' is not defined
```

Final remarks:
* the problem is not present when obfuscating with the default `--obf-code 1` setting
* the problem is not present if we change `from foo import models` to `from foo.models import Data` and use `Data` instead of `models.Data`

**Comment by jondy** [comment]
got it, thanks. I'll check it.

**Comment by jondy** [comment]
Fixed it in v8.2.3
